sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala

```diff
@@ -440,9 +441,16 @@ abstract class HadoopFsRelation private[sql](maybePartitionSpec: Option[Partitio
           val hdfsPath = new Path(path)
           val fs = hdfsPath.getFileSystem(hadoopConf)
           val qualified = hdfsPath.makeQualified(fs.getUri, fs.getWorkingDirectory)
-
-          logInfo(s"Listing $qualified on driver")
-          Try(fs.listStatus(qualified)).getOrElse(Array.empty)
+          logInfo(s"Listing $qualified on dr iver")
+          // Dummy jobconf to get to the pathFilter defined in configuration
+          val jobConf = new JobConf(hadoopConf, this.getClass())
+          val pathFilter = FileInputFormat.getInputPathFilter(jobConf)
+          if (pathFilter != null) {
+            Try(fs.listStatus(qualified, pathFilter)).getOrElse(Array.empty)
```

Comment:
@yhuai Thanks for taking a look and your comments. Yin, actually i am not sure about the usage of Try.  Is it because fs.listStatus can potentially throw an exception and subsequent call to getOrElse ? I just followed the existing logic .. Please let me know what you think ..